intake/auth/secret.py

```diff
@@ -0,0 +1,34 @@
+
+from .base import BasicAuth
+import uuid
+
+
+class SecretAuth(BasicAuth):
+    """A very simple auth mechanism using a shared secret
+
+    Parameters
+    ----------
+    secret: str
+        The string that must be matched in the requests. If None, a random UUID
+        is generated.
+    key: str
+        Header entry in which to seek the secret
+    """
+
+    def __init__(self, secret=None, key='intake-secret'):
+        if secret is None:
+            secret = uuid.uuid1().hex
```

Comment:
Does it make sense to autogenerate a secret key?  There's no way to even know what it is and communicate it to the client.  I'd rather raise an exception for not setting the `secret` argument.

Reply:
I am thinking about it like the jupyter token, something that could be printed to the console or stored at runtime. I would suggest actually that there is a log statement here to record the new random value.

Reply:
Ah, I see.  That makes more sense, although I don't think the usage model is that similar to Jupyter.  The intake server is likely to be started by a different person than the client, but I'm fine with either failing or picking a secret and logging it to the console.